### ISSUE TITLE
Consider other template engines

### ISSUE THREAD

**Issue by belak** [base16-builder/base16-builder#131]
One of the reasons the migration from the old base16 repos was so painful was because of the language switch. This is largely due to using language-specific template languages.

How possible would it be to use something like mustache which is very simple and has good bindings for most languages? Are there any other template languages which would be a better fit?

This would make it possible to build tooling in other languages without relying on the builder staying the same. It would also go a long way to splitting templates out into a separate repo (or repos) so not all the templates need to be maintained in a monolithic repo.

This idea was born out of a frustration that some features of the old builder useful for package maintainers were removed. One of the nice things about the old builder was that it was easier to script- you could (as an example) tell it to generate all emacs templates. This could then be published in a new repo as a package for emacs. That is also (somewhat) a separate issue, but this was one of the simple ideas I came up with which might help deal with that, so when there are disagreements, the base16 ecosystem (if you can call it that) doesn't get fragmented further.

**Comment by aloisdg** [comment]
_tl;dr: come talk in #126_

----

> One of the reasons the migration from the old base16 repos was so painful was because of the language switch. This is largely due to using language-specific template languages.

Agreed. The language-specific template language is hassle. Not quite a problem, but we want to improve this. We would love to share templates and schemes easily. We would love to see them purely language agnostic.

> How possible would it be to use something like mustache which is very simple and has good bindings for most languages?

I never use Mustache. Can we run code in it? How its handle? The main problem with templating engine is to deal with template like [smartftp](https://github.com/base16-builder/base16-builder/blob/master/db/templates/smartftp/dark.ejs):

```js
<%- base["00"]["hex"].match(/.{2}/g).reverse().join('') %>
```

An alternative would be to move all the logic from here to the builder. It is pain the original project nor b16b took.

> Are there any other template languages which would be a better fit?

@alexbooker started with nunjucks. He tried swig too. Finally, we move to ejs (Alex did all the job!). For reference see #67

> One of the nice things about the old builder was that it was easier to script- you could (as an example) tell it to generate all emacs templates.

We talk a lot about this with Alex and we agreed that the builder MUST print to `stdout` to follow the unix way. The generation of a group of template wasn't abandoned, but we decide that it belongs to a separate project (schemer is the result of the same way of thinking).

**Comment by belak** [comment]
Alright, I'll drop this in favor of #126.
